models/instance.go

```diff
@@ -98,3 +97,19 @@ func ValidateStateFilter(filterList []string) error {
 
 	return nil
 }
+
+// ValidateInstanceState checks the list of instance states from a whitelist
+func ValidateInstanceState(state string) error {
+	var invalidInstantStateValues []string
```

Comment:
Not sure why you need a slice of strings here, could this function be simplified to something like 
```
if _, ok := validStates[state]; !ok {
		err := fmt.Errorf("Bad request - invalid filter state value: %v", state)
		return err
}
return nil
```

Reply:
There is a user case on the web to search for a list of instances based on more than one state; and hence there could be multiple bad states the user has entered.